homeassistant/components/switchbot/sensor.py

```diff
@@ -93,4 +110,9 @@ def __init__(
     @property
     def native_value(self) -> str:
         """Return the state of the sensor."""
+        if self._sensor == "temp" and self.data["data"].get("temp"):
+            if self.data["data"].get("fahrenheit"):
+                return self.data["data"]["temp"].get("f")
+            return self.data["data"]["temp"].get("c")
+
```

Comment:
I think there might be a cleaner way to handle nested keys via the SensorEntityDescription "key=" value. Just need to test a few combinations.